src/components/views/rooms/MessageActions.js

```diff
@@ -186,39 +194,103 @@ function VideoCallButton(props) {
     };
 
     return <AccessibleButton className="mx_MessageComposer_button mx_MessageComposer_videocall"
+        tabIndex={props.tabIndex}
         onClick={onCallClick}
         title={_t('Video call')}
     />;
 }
 
 VideoCallButton.propTypes = {
     roomId: PropTypes.string.isRequired,
+    tabIndex: PropTypes.number,
 };
 
+/**
+ * This component implements the Toolbar design pattern from the WAI-ARIA
+ * Authoring Practices guidelines.
+ *
+ * https://www.w3.org/TR/wai-aria-practices-1.1/#toolbar
+ */
 export default function MessageActions(props) {
     const callInProgress = props.callState && props.callState !== 'ended';
+    const elRef = useRef(null);
+    const [focused, setFocused] = useState('upload');
+    const tabIndex = (target) => target === focused ? 0 : -1;
     const controls = [
-        <UploadButton key="upload" roomId={props.room.roomId} />,
-        <EmojiButton key="emoji" addEmoji={addEmoji} />,
-        <Stickerpicker key="sticker" room={props.room} />,
+        <UploadButton key="upload"
+                      tabIndex={tabIndex('upload')}
+                      roomId={props.room.roomId} />,
+        <EmojiButton key="emoji"
+                     tabIndex={tabIndex('emoji')}
+                     addEmoji={addEmoji} />,
+        <Stickerpicker key="sticker"
+                       tabIndex={tabIndex('sticker')}
+                       room={props.room} />,
```

Comment:
our multi-line props style is to do 4 space indent on the line after the component opener